src/audio/dcblock/dcblock_generic.c

```diff
@@ -0,0 +1,129 @@
+#include <stdint.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/dcblock/dcblock.h>
+
+/**
+ *
+ * Genereric processing function. Input is 32 bits.
+ *
+ */
+static int32_t dcblock_generic(struct dcblock_state *dcblock, int32_t x)
+{
+	int32_t x_prev = dcblock->x_prev;
+	int32_t y_prev = dcblock->y_prev;
+	int64_t R = dcblock->R_coeff;
+
+	int64_t out = x - x_prev + Q_SHIFT_RND(R * y_prev, 62, 31);
+
+	y_prev = sat_int32(out);
+	x_prev = x;
+
+	dcblock->x_prev = x_prev;
```

Comment:
You could eliminate lines lines 19-20 by writing them directly here to dcblock. Maybe the optimizer does it but it would at least look more compact.